extensions/typescript-language-features/src/languageFeatures/diagnostics.ts

```diff
@@ -148,6 +150,76 @@ class DiagnosticSettings {
 	}
 }
 
+class DiagnosticsTelemetryManager extends Disposable {
+
+	private readonly _diagnosticCodesMap = new Map<number, number>();
+	private readonly _diagnosticSnapshotsMap = new Map<vscode.Uri, readonly vscode.Diagnostic[]>();
```

Comment:
We can't use `vscode.Uri` as a key in this map because that would make it that two vscode.Uri instances which are equal would not have the same diagnostics. You can create a Map<string, ...> and use uri.toString().